src/index.js

```diff
@@ -1,39 +1,239 @@
+    edit.addEventListener('click', (e) => {
+      if (e.target.classList.contains('editInputShow')) {
+        const card = e.target.parentElement;
+        const input = card.querySelector('.editInput');
+        const item = card.querySelector('.item');
+        const deleteIcon = card.querySelector('.clear');
+        const editBtn = card.querySelector('.edit-btn');
+
+        editBtn.classList.remove('editInputShow');
+        input.classList.add('hide');
+        deleteIcon.classList.add('clearShow');
+        input.classList.add('editInputShow');
+        item.classList.add('hide');
+        editBtn.classList.add('editInputShow');
+        edit.classList.add('hide');
+        edit.classList.remove('editInputShow');
+      }
+    });
+  });
+
+  const add = document.getElementById('add-btn');
+  const txtInput = document.querySelector('.txt-input');
+  add.addEventListener('click', () => {
+    const item = txtInput.value.trim();
+    if (item) {
+      txtInput.value = '';
+      const todos = !localStorage.getItem('todos')
+        ? []
+        : JSON.parse(localStorage.getItem('todos'));
+      const currentTodo = {
+        id: todos.length + 1,
+        item,
+        completed: false,
+      };
+      addTodo([currentTodo]);
+      todos.push(currentTodo);
+      localStorage.setItem('todos', JSON.stringify(todos));
+    }
+    txtInput.focus();
+  });
+
+  document.querySelectorAll('.editInput').forEach((input) => {
+    input.addEventListener('keydown', (e) => {
+      const item = input.value.trim();
+      if (e.keyCode === 13) {
+        if (item) {
+          const todos = JSON.parse(localStorage.getItem('todos'));
+          const currentTodo = todos.find(
+            (todo) => parseInt(todo.id, 10) === parseInt(input.dataset.id, 10),
+          );
+
+          editTodo(todos.indexOf(currentTodo) + 1, item);
+        }
+      }
+    });
+  });
+
+  txtInput.addEventListener('keydown', (e) => {
+    if (e.keyCode === 13) {
+      add.click();
+    }
+  });
+
+  document.getElementById('clear-completed').addEventListener('click', () => {
+    const deleteIndexes = [];
+    document.querySelectorAll('.card.checked').forEach((card) => {
+      deleteIndexes.push(
+        [...document.querySelectorAll('.todos .card')].indexOf(card),
+      );
+      card.classList.add('fall');
+      card.addEventListener('animationend', () => {
+        setTimeout(() => {
+          card.remove();
+        }, 100);
+      });
+    });
+    removeManyTodo(deleteIndexes);
+  });
+}
+
+const updateIndex = (arr) => {
+  arr.forEach((task, index) => {
+    task.id = index + 1;
+  });
 };
 
-export default renderlists();
+const stateTodo = (index, completed) => {
+  const todos = JSON.parse(localStorage.getItem('todos'));
+  todos[index].completed = completed;
+  localStorage.setItem('todos', JSON.stringify(todos));
+};
+
+const removeTodo = (index) => {
+  const todos = JSON.parse(localStorage.getItem('todos'));
+  todos.splice(index, 1);
```

Comment:
- [Optional] Please let's avoid using methods that mutate the original array like splice(), sort(), reverse(), push(), unshift() etc and make use of methods like filter(), concat(), slice() etc.